diff --git a/dist/runtime/web/color-scheme.js b/dist/runtime/web/color-scheme.js
index c8fb1d2d9eb59204349cf80aeb56cbc281e81771..b80b72fe6a7d5fd9cbaeea85e5c943be6f36f9d0 100644
--- a/dist/runtime/web/color-scheme.js
+++ b/dist/runtime/web/color-scheme.js
@@ -33,6 +33,7 @@ else if ("window" in globalThis) {
         const flags = darkModeFlag.split(" ");
         darkMode = flags[0];
         darkModeValue = flags[1];
+        if (darkMode === "media") return;
         exports.colorScheme.set(globalThis.window.document.documentElement.classList.contains(darkModeValue)
             ? "dark"
             : "system");
