diff --git a/lib/commonjs/commons/utils/getSwipeDirection.js b/lib/commonjs/commons/utils/getSwipeDirection.js
index 20c248421a9a9ba3f22aee4631204623575fffe1..f49e2ac04a9ce62ded71dbde1853b343fc26a22e 100644
--- a/lib/commonjs/commons/utils/getSwipeDirection.js
+++ b/lib/commonjs/commons/utils/getSwipeDirection.js
@@ -4,8 +4,8 @@ Object.defineProperty(exports, "__esModule", {
   value: true
 });
 exports.getSwipeDirection = void 0;
-const SWIPE_TIME = 175;
-const SWIPE_VELOCITY = 500;
+const SWIPE_TIME = 350;
+const SWIPE_VELOCITY = 250;
 const SWIPE_DISTANCE = 20;
 const getSwipeDirection = (e, options) => {
   'worklet';
diff --git a/lib/commonjs/commons/utils/snapPoint.js b/lib/commonjs/commons/utils/snapPoint.js
index db49d341501cacd9cade69318615840587684bb8..ed31b951036fa9c8fe0b8a4a62e0d973f4c328fd 100644
--- a/lib/commonjs/commons/utils/snapPoint.js
+++ b/lib/commonjs/commons/utils/snapPoint.js
@@ -7,7 +7,7 @@ exports.snapPoint = void 0;
 const snapPoint = (value, velocity, points) => {
   'worklet';
 
-  const point = value + 0.05 * velocity;
+  const point = value + 0.15 * velocity;
   const deltas = points.map(p => Math.abs(point - p));
   const minDelta = Math.min.apply(null, deltas);
   return points.filter(p => Math.abs(point - p) === minDelta)[0];
diff --git a/lib/commonjs/components/gallery/GalleryGestureHandler.js b/lib/commonjs/components/gallery/GalleryGestureHandler.js
index f2d7c9c7a2e2a8b82aa0a92e563224cb3327c000..94a5beabdab3ee170d740afd50d682f427f67e02 100644
--- a/lib/commonjs/components/gallery/GalleryGestureHandler.js
+++ b/lib/commonjs/components/gallery/GalleryGestureHandler.js
@@ -20,8 +20,8 @@ function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return
 function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
 const minScale = 1;
 const config = {
-  duration: 300,
-  easing: _reactNativeReanimated.Easing.linear
+  duration: 250,
+  easing: _reactNativeReanimated.Easing.out(_reactNativeReanimated.Easing.cubic)
 };
 /*
  * Pinchable views are really heavy components, therefore in order to maximize performance
diff --git a/lib/module/commons/utils/getSwipeDirection.js b/lib/module/commons/utils/getSwipeDirection.js
index 7da596ac92d55319caa8b12100cc8317af3ea8fa..e6cc279f856acb7f8760d0ed7dd85e25b5ad1580 100644
--- a/lib/module/commons/utils/getSwipeDirection.js
+++ b/lib/module/commons/utils/getSwipeDirection.js
@@ -1,5 +1,5 @@
-const SWIPE_TIME = 175;
-const SWIPE_VELOCITY = 500;
+const SWIPE_TIME = 350;
+const SWIPE_VELOCITY = 250;
 const SWIPE_DISTANCE = 20;
 export const getSwipeDirection = (e, options) => {
   'worklet';
diff --git a/lib/module/commons/utils/snapPoint.js b/lib/module/commons/utils/snapPoint.js
index 6ea170c0a2a70e4ea25b1635cb3f75bce1395add..ef564e82ee3aff1713c558fab3b9617582d7089a 100644
--- a/lib/module/commons/utils/snapPoint.js
+++ b/lib/module/commons/utils/snapPoint.js
@@ -1,7 +1,7 @@
 export const snapPoint = (value, velocity, points) => {
   'worklet';
 
-  const point = value + 0.05 * velocity;
+  const point = value + 0.15 * velocity;
   const deltas = points.map(p => Math.abs(point - p));
   const minDelta = Math.min.apply(null, deltas);
   return points.filter(p => Math.abs(point - p) === minDelta)[0];
diff --git a/lib/module/components/gallery/GalleryGestureHandler.js b/lib/module/components/gallery/GalleryGestureHandler.js
index 0e53798e177dc64a3f7f74844e29317d8cbfcbfc..6f2a6fbb97a42630a6fe03c0924e23751e604e97 100644
--- a/lib/module/components/gallery/GalleryGestureHandler.js
+++ b/lib/module/components/gallery/GalleryGestureHandler.js
@@ -12,8 +12,8 @@ import { GalleryContext } from './context';
 import { getScrollPosition } from '../../commons/utils/getScrollPosition';
 const minScale = 1;
 const config = {
-  duration: 300,
-  easing: Easing.linear
+  duration: 250,
+  easing: Easing.out(Easing.cubic)
 };
 /*
  * Pinchable views are really heavy components, therefore in order to maximize performance
diff --git a/src/commons/utils/getSwipeDirection.ts b/src/commons/utils/getSwipeDirection.ts
index a3599652cf52f456d23f40ac70b1546786adb36b..30b3f2c54a7df18c13920b2cf449e715825c0afd 100644
--- a/src/commons/utils/getSwipeDirection.ts
+++ b/src/commons/utils/getSwipeDirection.ts
@@ -7,8 +7,8 @@ type SwipeOptions = {
   translate: Vector<number>;
 };
 
-const SWIPE_TIME = 175;
-const SWIPE_VELOCITY = 500;
+const SWIPE_TIME = 350;
+const SWIPE_VELOCITY = 250;
 const SWIPE_DISTANCE = 20;
 
 export const getSwipeDirection = (
diff --git a/src/commons/utils/snapPoint.ts b/src/commons/utils/snapPoint.ts
index f8a5b7f4305a3113cdd0ea647fc94e036a35fdc4..5af0f6a978da95caf688c3e0b05bdeaf33987a1b 100644
--- a/src/commons/utils/snapPoint.ts
+++ b/src/commons/utils/snapPoint.ts
@@ -4,7 +4,7 @@ export const snapPoint = (
   points: ReadonlyArray<number>
 ): number => {
   'worklet';
-  const point = value + 0.05 * velocity;
+  const point = value + 0.15 * velocity;
   const deltas = points.map((p) => Math.abs(point - p));
   const minDelta = Math.min.apply(null, deltas);
   return points.filter((p) => Math.abs(point - p) === minDelta)[0]!;
diff --git a/src/components/gallery/GalleryGestureHandler.tsx b/src/components/gallery/GalleryGestureHandler.tsx
index f04c32c221f4e3f8d5ab8606ada3cc881a40f0c4..61bec48e934c3f08b77b0ae8e38262e995df3306 100644
--- a/src/components/gallery/GalleryGestureHandler.tsx
+++ b/src/components/gallery/GalleryGestureHandler.tsx
@@ -33,7 +33,7 @@ import { type GalleryProps } from './types';
 import { getScrollPosition } from '../../commons/utils/getScrollPosition';
 
 const minScale = 1;
-const config = { duration: 300, easing: Easing.linear };
+const config = { duration: 250, easing: Easing.out(Easing.cubic) };
 
 type GalleryGestureHandlerProps = {
   length: number;
